kmod/src/block.c

```diff
@@ -1160,12 +1168,11 @@ static int block_shrink(struct shrinker *shrink, struct shrink_control *sc)
 
 	rhashtable_walk_stop(&iter);
 	rhashtable_walk_exit(&iter);
-out:
+
 	if (stop)
 		return -1;
```

Comment:
Returning -1 as an unsigned long should raise eyebrows.  This is the semantics of the old single shrink.  The new scan should return SHRINK_STOP in this case.  The compat wrapper can see that and return -1 for the single int shrink function.